acbuild/acbuild.go

```diff
@@ -156,6 +163,52 @@ func stdout(format string, a ...interface{}) {
 	fmt.Fprintln(os.Stdout, strings.TrimSuffix(out, "\n"))
 }
 
+func getLock() (*os.File, error) {
+	ex, err := util.Exists(path.Join(contextpath, workprefix))
+	if err != nil {
+		return nil, err
+	}
+	if !ex {
+		return nil, fmt.Errorf("build not in progress in this working dir - try \"acbuild begin\"")
+	}
+
+	lockfile, err := os.OpenFile(lockpath(), os.O_CREATE|os.O_RDWR, 0644)
+	if err != nil {
+		return nil, err
+	}
+
+	err = syscall.Flock(int(lockfile.Fd()), syscall.LOCK_EX|syscall.LOCK_NB)
+	if err != nil {
+		if err == syscall.EWOULDBLOCK {
+			return nil, fmt.Errorf("lock already held - is another acbuild running in this working dir?")
+		} else {
```

Comment:
Nit, it's preferred style to just do short-circuit return - so, remove the "else"